rfd/xxxx-azure-cosmos-integration.md

```diff
@@ -0,0 +1,165 @@
+---
+authors: Gabriel Corado (gabriel.oliveira@goteleport.com)
+state: draft
+---
+
+# RFD xxxx - Azure CosmosDB integration
+
+## Required approvers
+
+* Engineering: @smallinsky && @r0mant
+* Product: @xinding33
+
+## What
+
+CosmosDB for MongoDB and Cassandra integration in Teleport Database Access.
+
+## Why
+
+Users will be able to access their CosmosDB databases accounts (MongoDB and
+Cassandra APIs) with Teleport Database Access taking advantage of it like an
+audit logging and access controls.
+
+## Details
+
+### Authentication
+
+Teleport already supports MongoDB and Cassandra protocols. The only necessary
+change is in the authentication options.
+
+To connect to CosmosDB, it has database account keys, which are used as
```

Comment:
Are these the primary / secondary keys described here: https://learn.microsoft.com/en-us/azure/cosmos-db/secure-access-to-data?tabs=using-primary-key#primary-keys

If so this seems like a potentially concerning credential to use.  As I understand it this is an account wide credential rather than an application specific credential.

Were the other two authentication methods mentioned on that document considered?

Reply:
> Are these the primary / secondary keys described here: https://learn.microsoft.com/en-us/azure/cosmos-db/secure-access-to-data?tabs=using-primary-key#primary-keys

Yes.

> Were the other two authentication methods mentioned on that document considered?

It is worth mentioning that different CosmosDB APIs (such as MongoDB and Cassandra) can/will implement them differently.

For **Resource tokens**, it only works with [CosmosDB NoSQL API](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/query/getting-started), which uses a REST API (with SQL queries) instead of a database protocol.

For the **Role-based access control (RBAC)** model Cassandra doesn't support it. For MongoDB, there is a section on the [last section of the RFD (Alternatives)](https://github.com/gravitational/teleport/blob/gabrielcorado/rfd-azure-cosmos/rfd/xxxx-azure-cosmos-integration.md#mongodb-rbac) about it.

Reply:
Thank you, I was not completely aware of that.  The performance issues described seems surprising to me.  Have we investigated these at all, or raised a support issue with Azure to help us investigate?

If possible, the RBAC auth still seems like a better option to me.  It allows our customers much more fine grained control on the access provided.  I also worry that using this account level token we may result in unexpected interactions with other tools.  For example, I don't think we could safely assume we could initiate a key rotation due to the risk that other tools may not have the list key's permission and instead require a manual update.

Let me know your thoughts, thank you @gabrielcorado!